build.gradle

```diff
@@ -1,70 +1,69 @@
+	id "architectury-plugin" version "3.4-SNAPSHOT"
+	id "dev.architectury.loom" version "1.4-SNAPSHOT" apply false
 }
 
-base.archivesName = project.archives_base_name
-version = project.mod_version
-group = project.maven_group
-if ("false".equalsIgnoreCase(System.getenv("IS_RELEASE"))) {
-	version += "-SNAPSHOT"
+architectury {
+	minecraft = rootProject.minecraft_version
 }
 
-dependencies {
-	minecraft "com.mojang:minecraft:${project.minecraft_version}"
-	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
-}
-
-processResources {
-	inputs.property "version", version
+subprojects {
+	apply plugin: "dev.architectury.loom"
 
-	filesMatching("fabric.mod.json") {
-		expand "version": version
+	dependencies {
+		minecraft "com.mojang:minecraft:${rootProject.minecraft_version}"
+		mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
 	}
-}
 
-// ensure that the encoding is set to UTF-8, no matter what the system default is
-// this fixes some edge cases with special characters not displaying correctly
-// see http://yodaconditions.net/blog/fix-for-java-file-encoding-problems-with-gradle.html
-tasks.withType(JavaCompile).configureEach {
-	options.encoding = "UTF-8"
+	// TODO: jar { from("LICENSE") }?
 }
 
-java {
-	sourceCompatibility = JavaVersion.VERSION_1_8
-	targetCompatibility = JavaVersion.VERSION_1_8
+allprojects {
+	apply plugin: "java"
+	apply plugin: "architectury-plugin"
+	apply plugin: "maven-publish"
 
-	// Loom will automatically attach sourcesJar to a RemapSourcesJar task and to the "build" task
-	// if it is present.
-	// If you remove this line, sources will not be generated.
-	withSourcesJar()
-}
-
-jar {
-	from("LICENSE") {
-		rename { "${it}_${project.archives_base_name}"}
+	base {
+		archivesName = rootProject.archives_base_name
 	}
-}
 
-// configure the maven publication
-publishing {
-	publications {
-		mavenJava(MavenPublication) {
-			from components.java
-		}
+	version = rootProject.mod_version
+	group = rootProject.maven_group
+	if ("false".equalsIgnoreCase(System.getenv("IS_RELEASE"))) {
+		version += "-SNAPSHOT"
 	}
 
-	// select the repositories you want to publish to
 	repositories {
-		maven {
-			url = version.endsWith("SNAPSHOT") ? "https://maven.fallenbreath.me/snapshots" : "https://maven.fallenbreath.me/releases"
-			credentials(PasswordCredentials) {
-				username = 'fallen'
-				password = System.getenv("FALLENS_MAVEN_TOKEN")
-			}
-			authentication {
-				basic(BasicAuthentication)
+		// Add repositories to retrieve artifacts from in here.
+		// You should only use this when depending on other mods because
+		// Loom adds the essential maven repositories to download Minecraft and libraries from automatically.
+		// See https://docs.gradle.org/current/userguide/declaring_repositories.html
+		// for more information about repositories.
+	}
+
+	tasks.withType(JavaCompile) {
+		options.encoding = "UTF-8"
+		options.release = 17
```

Comment:
will this affect java8 compatibility

Reply:
turns out, yes it does, but I've been able to resolve that